search-app/src/cmr/search/api/concepts_search.clj

```diff
@@ -366,13 +367,36 @@
   ;; no errors, return 204
   {:status 204})
 
+(defn pluralize-concept
+  "Return the plural form of a concept"
+  [concept]
+  (let [concept-name (name concept) last-letter (subs concept-name (- (count concept-name) 1))]
+    (if (= "y" last-letter)
+      (keyword (str (subs concept-name 0 (- (count concept-name) 1)) "ies"))
+      (keyword (str concept-name "s")))))
+
+(defn format-regex
+  "Adds the necessary string to each concept for a regex"
+  [concept] (str "(?" concept ")"))
+
+(def get-generics
+  "Retrieve the generics which were dynamically loaded, drop the keyword :generic"
+  (mapv pluralize-concept (drop 1 (keys concepts/generic-concept-types->concept-prefix))))
```

Comment:
I believe you can skip the dropping issue by using the base list of generics, rather than the prefixes function. So, change to (mapv pluralize-concept (keys cmr.common.config/approved-pipeline-documents))

I have not tested this but 99% sure.

Reply:
does this drop "drop" the "generic" value which is added in the list?

Reply:
Correct it drops the keyword :generic from the list which has (:generic, :grid, ....)

Reply:
it is undefined which is first though, nothing says it can't be last, can you remove it another way so that it is more clear which is the offending item?

Reply:
Using a different source approved-pipeline-documents I no longer use the drop